geonode/base/api/views.py

```diff
@@ -198,10 +198,26 @@ class OwnerViewSet(WithDynamicViewSetMixin, ListModelMixin, GenericViewSet):
     """
     authentication_classes = [SessionAuthentication, BasicAuthentication, OAuth2Authentication]
     permission_classes = [AllowAny, ]
-    queryset = get_user_model().objects.all().exclude(pk=-1)
     serializer_class = OwnerSerializer
     pagination_class = GeoNodeApiPagination
 
+    def get_queryset(self):
+        """
+        Filter users with atleast a resource
+        """
+        queryset = get_user_model().objects.all().exclude(pk=-1)
+        filter_options = {}
+        if self.request.query_params:
+            filter_options = {
+                'type_filter': self.request.query_params.get('type'),
+                'title_filter': self.request.query_params.get('title__icontains')
+                }
+        owner_ids = get_resources_with_perms(
```

Comment:
why don't you use a Subquery? Your solution will issue two DB queries, doesn't it?
https://docs.djangoproject.com/en/2.2/ref/models/expressions/#subquery-expressions

Reply:
updated